packages/sources/therundown/src/endpoint/events.ts

```diff
@@ -0,0 +1,49 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { ExecuteWithConfig, Config } from '@chainlink/types'
+
+export const NAME = 'events'
+
+const customParams = {
+  sportId: true,
+  date: true,
+  status: false
+}
+
+export interface ResponseSchema {
+  events: {
+    score: {
+      event_status: string
+    }
+  }[]
+}
+
+export const execute: ExecuteWithConfig<Config> = async (request, config) => {
+  const validator = new Validator(request, customParams)
+  if (validator.error) throw validator.error
+
+  const jobRunID = validator.validated.id
+  const sportId = validator.validated.data.sportId
+  const date = validator.validated.data.date
+  const status = validator.validated.data.status
+  const url = `/sports/${sportId}/events/${date}`
+
+  const reqConfig = {
+    ...config.api,
+    headers: {
+      ...config.api.headers,
+      'x-rapidapi-key': config.apiKey,
+    },
+    params: {
+      include: 'scores',
+    },
+    url,
+  }
+
+  const response = await Requester.request(reqConfig)
+  if (status !== undefined) {
```

Comment:
In the case that `status` _is_ undefined should we be throwing an error?

If so we could pass a callback to `Requester.request` with a custom validation check for this.